x-pack/plugins/lens/common/expressions/datatable/transpose_helpers.test.ts

```diff
@@ -179,22 +178,22 @@ describe('transpose_helpers', () => {
     expect(table.columns.map((c) => c.id)).toEqual([
       'bucket3',
       'A---D---metric1',
-      'B---D---metric1',
-      'C---D---metric1',
+      'A---D---metric2',
       'A---E---metric1',
-      'B---E---metric1',
-      'C---E---metric1',
+      'A---E---metric2',
       'A---F---metric1',
-      'B---F---metric1',
-      'C---F---metric1',
-      'A---D---metric2',
+      'A---F---metric2',
+      'B---D---metric1',
       'B---D---metric2',
-      'C---D---metric2',
-      'A---E---metric2',
+      'B---E---metric1',
       'B---E---metric2',
-      'C---E---metric2',
-      'A---F---metric2',
+      'B---F---metric1',
       'B---F---metric2',
+      'C---D---metric1',
+      'C---D---metric2',
+      'C---E---metric1',
+      'C---E---metric2',
+      'C---F---metric1',
       'C---F---metric2',
```

Comment:
Now sorting transposed table columns in the order they are displayed on the table, by value(s) then metric.